projects/illm/train.py

```diff
@@ -0,0 +1,375 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import hashlib
+import logging
+import math
+import pickle
+from pathlib import Path
+from typing import List, Optional, Tuple, Union
+from warnings import warn
+
+import hydra
+import lightning.pytorch as pl
+import torch
+import torch.nn as nn
+import wandb
+import yaml
+from gan_compression_module import GANCompressionModule
+from image_module import ImageModule
+from lightning.pytorch.callbacks import LearningRateMonitor, ModelCheckpoint
+from lightning.pytorch.loggers import WandbLogger
+from lightning.pytorch.strategies import DDPStrategy
+from omegaconf import DictConfig, OmegaConf
+from open_images_datamodule import OpenImagesDataModule
+from target_compression_module import TargetRateCompressionModule
+from torch import Tensor
+from torch.utils.data import DataLoader
+from torchvision.utils import make_grid
+
+from neuralcompression.loss_fn import TargetRateConfig
+from neuralcompression.models import HyperpriorAutoencoderBase
+
+LOGGER = logging.getLogger(__file__)
+
+
+class WandbImageCallback(pl.Callback):
```

Comment:
Note: The wandb part is to me a major block to using the code if it cannot be removed with just a single flag.

Reply:
I added a `try`/`except` for now so the code should potentially run if the user doesn't have it, but this should probably be configurable.